framework/src/modules/interoperability/mainchain/module.ts

```diff
@@ -261,15 +261,7 @@ export class MainchainInteroperabilityModule extends BaseInteroperabilityModule
 			throw new Error(`ownChainName must be equal to ${CHAIN_NAME_MAINCHAIN}.`);
 		}
 
-		// if chainInfos is empty, then ownChainNonce == 0
-		// If chainInfos is non-empty, ownChainNonce > 0
-		if (chainInfos.length === 0 && ownChainNonce !== BigInt(0)) {
-			throw new Error(`ownChainNonce must be 0 if chainInfos is empty.`);
-		} else if (chainInfos.length !== 0 && ownChainNonce <= 0) {
-			throw new Error(`ownChainNonce must be positive if chainInfos is not empty.`);
-		}
-
-		this._verifyChainInfos(ctx, chainInfos);
+		this._verifyChainInfos(ctx, chainInfos, ownChainNonce);
```

Comment:
Here I assume it's just a re-structuring change, by moving the nonce-length checks inside the `verifyChainInfos` function and there is no change in the logic. Is that right?